src/main/java/net/fabricmc/loom/configuration/providers/minecraft/MinecraftLibraryProvider.java

```diff
@@ -108,13 +108,7 @@ private void provideClientLibraries() {
 	}
 
 	private void provideServerLibraries() {
-		final BundleMetadata serverBundleMetadata = minecraftProvider.getServerBundleMetadata();
-
-		if (serverBundleMetadata == null) {
-			return;
-		}
-
-		final List<Library> libraries = MinecraftLibraryHelper.getServerLibraries(serverBundleMetadata);
+		final List<Library> libraries = MinecraftLibraryHelper.getServerLibraries(minecraftProvider.getServerBundleMetadata());
```

Comment:
Is this not always going to be empty for the older versions without the bundle metadata, or am I missing something?

Reply:
It is, but custom library processors can add their own libraries.

Reply:
I dont think this is the correct fix, and im not sure it actually fixes the issue. The current library processor API doesnt really expect you to be able to add libraries, its very much for changing the vanilla ones. What is your usecase? Can you not just add the library in the gradle way?

Reply:
I have a plugin that upgrades or adds libraries depending on the MC version. For example, MC didn't depend on Log4j until some 1.7 snapshot, so for earlier versions it adds that dependencies, for later versions it upgrades it to a newer version.

Reply:
Ok, I see yeah. Sorry yes that is possible.

This change is correct, but im not super happy with how it changes getServerLibraries. I think you should change it all in `provideServerLibraries`. Something as simple as:

```java
private void provideServerLibraries() {
	final BundleMetadata serverBundleMetadata = minecraftProvider.getServerBundleMetadata();
	final List<Library> libraries = serverBundleMetadata != null ? MinecraftLibraryHelper.getServerLibraries(serverBundleMetadata) : Collections.emptyList();
	final List<Library> processLibraries = processLibraries(libraries);
	processLibraries.forEach(this::applyServerLibrary);
}
```

Reply:
alrighty, done